src/main.rs

```diff
@@ -164,18 +164,22 @@ fn main() -> std::io::Result<()> {
 }
 
 fn write_once(buffer: &[u8]) -> std::io::Result<Duration> {
-    let mut write_time = Duration::new(0, 0);
+    let start = Instant::now();
     {
         let mut file = File::create("test").expect("Can't open test file");
 
         for _ in 0..TOTAL_SIZE_MB / BUF_SIZE_MB {
-            write_time += prof! {
-                file.write_all(buffer)?;
-            };
+            file.write_all(buffer)?;
+            // make sure the data is synced with the disk as the kernel performs
+            // write buffering
+            //
+            // TODO Open the file in O_DSYNC instead to avoid the additional syscall
+            file.sync_data()?;
             print!(".");
```

Comment:
this would stay out of the `prof!` block as before.

Reply:
The sync_data() must be included in the `prof!` block as this is when the os is actually transferring the data to the disk.